core/templates/dev/head/domain/collection/CollectionNodeObjectFactory.js

```diff
@@ -103,6 +103,12 @@ oppia.factory('CollectionNodeObjectFactory', [
         explorationSummaryBackendObject);
     };
 
+    CollectionNode.prototype.getCapitalizedObjective = function(
+        explorationObjective) {
```

Comment:
One more design issue -- this shouldn't take in an objective as an argument; instead, it should make use of the objective in this._explorationSummaryObject. Otherwise you potentially have two different sources of truth. (Think about it like this: someone external is asking this collection node for their 'capitalized objective', and this is something that the collection node should be able to derive on their own from the objective that they already 'know'.)